src/System.Private.Xml/src/System/Xml/NameTable.cs

```diff
@@ -176,6 +175,25 @@ public override string Get(char[] key, int start, int len)
             return null;
         }
 
+        internal string GetOrAddEntry(string str, int hashCode)
+        {
+            for (Entry e = _entries[hashCode & _mask]; e != null; e = e.next)
```

Comment:
Since this is internal func only called by new code, would it make sense to add a null check for `str` here in case for whatever reason we added null (i.e. field initialized in different order than we thought)

Reply:
This method should be used only when adding "constant strings" for which we stored the hashes, I don't see much value in adding a check for null here.